modules/apps/transfer/types/transfer_authorization.go

```diff
@@ -131,6 +131,16 @@ func (a TransferAuthorization) ValidateBasic() error {
 			}
 			found[allocation.AllowList[i]] = true
 		}
+
+		if len(allocation.AllowedPacketData) > 0 && allocation.AllowedPacketData[0] != AllowAllPacketDataKeys {
+			jsonObject := make(map[string]interface{})
+			for _, elem := range allocation.AllowedPacketData {
+				err := json.Unmarshal([]byte(elem), &jsonObject)
+				if err != nil {
+					return errorsmod.Wrapf(ErrInvalidAuthorization, "allowed packet data contains a non JSON-encoded string: %s", elem)
+				}
+			}
+		}
```

Comment:
This checks that if the wildcard is not used, then all elements should be JSON-encoded strings.